src/libs/highlevel/elektra.c

```diff
@@ -0,0 +1,233 @@
+		return NULL;
+	}
+
+	Elektra * const elektra = elektraCalloc (sizeof (struct _Elektra));
+	elektra->kdb = kdb;
+	elektra->parentKey = parentKey;
+	elektra->config = config;
+	elektra->lookupKey = keyNew (NULL, KEY_END);
+	elektra->enforceType = true;
+	elektra->fatalErrorHandler = &defaultFatalErrorHandler;
+
+	return elektra;
+}
+
+/**
+ * Promote an ElektraError to fatal and call the fatal error handler.
+ *
+ * @param elektra    Elektra instance whose fatal error handler shall be used.
+ * @param fatalError The error that will be raised.
+ */
+void elektraFatalError (Elektra * elektra, ElektraError * fatalError)
+{
+	fatalError->severity = ELEKTRA_ERROR_SEVERITY_FATAL;
+	elektra->fatalErrorHandler (fatalError);
+}
+
+/**
+ * Sets the fatal error handler that will be called, whenever a fatal error occurs.
+ *
+ * Errors occurring in a function, which does not take a pointer to ElektraError,
+ * are always considered fatal.
+ *
+ * If this function returns, i.e. it does not call exit() or interrupt the thread of
+ * execution in some other way, the behaviour of the function from which the error
+ * originated is generally undefined.
+ *
+ * @param elektra           An Elektra instance.
+ * @param fatalErrorHandler The error handler that will be used henceforth.
+ */
+void elektraFatalErrorHandler (Elektra * elektra, ElektraErrorHandler fatalErrorHandler)
+{
+	elektra->fatalErrorHandler = fatalErrorHandler;
+}
+
+/**
+ * Sets whether type metadata should be enforced by this Elektra instance.
+ *
+ * Even if type metadata is not enforce, setting a value will always set the
+ * corresponding type metadata value too.
+ *
+ * @param elektra             An Elektra instance.
+ * @param enforceTypeMetadata Set to #true to enforce type metadata.
+ */
+void elektraEnforceTypeMetadata (Elektra * elektra, bool enforceTypeMetadata)
+{
+	elektra->enforceType = enforceTypeMetadata;
+}
+
+/**
+ * Releases all ressources used by the given elektra instance. The elektra instance must not be used anymore after calling this.
+ * @param elektra An Elektra instance.
+ */
+void elektraClose (Elektra * elektra)
+{
+	kdbClose (elektra->kdb, elektra->parentKey);
+	keyDel (elektra->parentKey);
+	ksDel (elektra->config);
+	keyDel (elektra->lookupKey);
+
+	elektraFree (elektra);
+}
+
+/**
+ * @}
+ */
+
+// Private definitions
+
+KDBType KDB_TYPE_STRING = "string";
+KDBType KDB_TYPE_BOOLEAN = "boolean";
+KDBType KDB_TYPE_CHAR = "char";
+KDBType KDB_TYPE_OCTET = "octet";
+KDBType KDB_TYPE_SHORT = "short";
+KDBType KDB_TYPE_UNSIGNED_SHORT = "unsigned_short";
+KDBType KDB_TYPE_LONG = "long";
+KDBType KDB_TYPE_UNSIGNED_LONG = "unsigned_long";
+KDBType KDB_TYPE_LONG_LONG = "long_long";
+KDBType KDB_TYPE_UNSIGNED_LONG_LONG = "unsigned_long_long";
+KDBType KDB_TYPE_FLOAT = "float";
+KDBType KDB_TYPE_LONG_DOUBLE = "long_double";
+KDBType KDB_TYPE_DOUBLE = "double";
+KDBType KDB_TYPE_ENUM = "enum";
+
+void elektraSetLookupKey (Elektra * elektra, const char * name)
```

Comment:
It is declared in `elektra_private.h` because multiple source files use this. The alternative would be to move everything into one source file and declare this function `static` in there.